scripts/eosio_build_darwin.sh

```diff
@@ -1,406 +1,311 @@
-		echo "Exiting now."
-		exit 1
-	fi
+printf "\\nOS name: ${OS_NAME}\\n"
+printf "OS Version: ${OS_VER}\\n"
+printf "CPU speed: ${CPU_SPEED}Mhz\\n"
+printf "CPU cores: %s\\n" "${CPU_CORE}"
+printf "Physical Memory: ${MEM_GIG} Gbytes\\n"
+printf "Disk install: ${DISK_INSTALL}\\n"
+printf "Disk space total: ${DISK_TOTAL}G\\n"
+printf "Disk space available: ${DISK_AVAIL}G\\n"
 
-	if [ "${OS_MIN}" -lt 12 ]; then
-		echo "You must be running Mac OS 10.12.x or higher to install EOSIO."
-		echo "Exiting now."
-		exit 1
-	fi
+if [ "${MEM_GIG}" -lt 7 ]; then
+	echo "Your system must have 7 or more Gigabytes of physical memory installed."
+	echo "Exiting now."
+	exit 1
+fi
 
-	if [ "${DISK_AVAIL}" -lt "$DISK_MIN" ]; then
-		echo "You must have at least ${DISK_MIN}GB of available storage to install EOSIO."
-		echo "Exiting now."
-		exit 1
-	fi
+if [ "${OS_MIN}" -lt 12 ]; then
+	echo "You must be running Mac OS 10.12.x or higher to install EOSIO."
+	echo "Exiting now."
+	exit 1
+fi
 
-	printf "\\tChecking xcode-select installation\\n"
-	if ! XCODESELECT=$( command -v xcode-select)
-	then
-		printf "\\n\\tXCode must be installed in order to proceed.\\n\\n"
-		printf "\\tExiting now.\\n"
-		exit 1
-	fi
+if [ "${DISK_AVAIL}" -lt "$DISK_MIN" ]; then
+	echo "You must have at least ${DISK_MIN}GB of available storage to install EOSIO."
+	echo "Exiting now."
+	exit 1
+fi
 
-	printf "\\txcode-select installation found @ \\n"
-	printf "\\t%s \\n\\n" "${XCODESELECT}"
+printf "\\n"
 
-	printf "\\tChecking Ruby installation.\\n"
-	if ! RUBY=$( command -v ruby)
-	then
-		printf "\\nRuby must be installed in order to proceed.\\n\\n"
-		printf "\\tExiting now.\\n"
-		exit 1
-	fi
+printf "Checking xcode-select installation...\\n"
+if ! XCODESELECT=$( command -v xcode-select)
+then
+	printf " - XCode must be installed in order to proceed!\\n\\n"
+	exit 1
+fi
+printf " - XCode installation found @ ${XCODESELECT}\\n"
 
-	printf "\\tRuby installation found @ \\n"
-	printf "\\t%s \\n\\n" "${RUBY}"
+printf "Checking Ruby installation...\\n"
+if ! RUBY=$( command -v ruby)
+then
+	printf " - Ruby must be installed in order to proceed!\\n"
+	exit 1
+fi
+printf " - Ruby installation found @ ${RUBY}\\n"
 
-	printf "\\tChecking Home Brew installation\\n"
-	if ! BREW=$( command -v brew )
-	then
-		printf "\\tHomebrew must be installed to compile EOS.IO\\n\\n"
-		printf "\\tDo you wish to install Home Brew?\\n"
-		select yn in "Yes" "No"; do
-			case "${yn}" in
-				[Yy]* )
-				"${XCODESELECT}" --install 2>/dev/null;
-				if ! "${RUBY}" -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
-				then
-					echo "Unable to install homebrew at this time. Exiting now."
-					exit 1;
-				else
-					BREW=$( command -v brew )
-				fi
-				break;;
-				[Nn]* ) echo "User aborted homebrew installation. Exiting now.";
-						exit 1;;
-				* ) echo "Please enter 1 for yes or 2 for no.";;
-			esac
-		done
-	fi
+printf "Checking CMAKE installation...\\n"
```

Comment:
No change in my opinion from last PR. Don't believe we need to build cmake from source on platforms that give us a new enough cmake

Reply:
We could just remove this entirely, right? It was only for Darwin originally, and never set for Linux, but from what I can tell if LLVM_DIR isn't set it looks under what I've specified in the ENV vars we have users set.

Reply:
Totally understand the concern. Unfortunately brew's cmake (or any brew packages) should not be relied upon for build/testing, as it is subject to change without warning and differ slightly (arguments they use to install) from the linux packages. In order to ensure confidence, reliability, and consistency between distros, we should be source installing as much as we can. 

My big concerns are around: What if brew upgrades the version of cmake that differs from the linux versions? If that breaks the mac build process for users of the scripts, and we don't notice it, do we just rely on them to notify us (create a bad user experience)?

It's possible that we can just continue to use the homebrew-eosio repo for our own tap, but then we (likely the blockchain team) have to maintain the formulas, which costs money and time for engineers/developers. Through uniformity (building cmake from source for all distros), we can eliminate many types of waste for everyone involved.

This also applies to llvm and other packages we're not source installing, but I'm told we're moving away from needing them in the future anyway.

Reply:
This is a slippery slope. Are you going to start compiling openssl and gmp too? Are you going to lock the build down to a specific Xcode version? The line has to be drawn somewhere and historically it has been drawn at the clear point where version differences impact nodeos data file compatibility.

We’re not building an embedded or proprietary software system, we’re building open source software that uses open source tools and components. Platform and tool diversity is a part of life and ignoring it reflects poorly — much more complex open source software doesn’t need to lock it self down to a specific cmake version for example.

Reply:
Do you think the current if statement to check if cmake already exists solves most of your concerns?
```CMAKE=$(command -v cmake 2>/dev/null)
if [ -z $CMAKE ]; then```

Reply:
Believe this was addressed after discussions about homebrew in Dawn channel today.